nexus/users/models.py

```diff
@@ -30,21 +32,49 @@ def create_superuser(self, email: str, password: str, **extra_fields):
 
 
 class User(AbstractBaseUser, UUIDModel, PermissionsMixin):
-    first_name = models.CharField(_('First Name'), max_length=120, blank=True)
-    last_name = models.CharField(_('Last Name'), max_length=120, blank=True)
+
+    GENDER_CHOICES = Choices(
+        ('NOT_SELECTED', 'NA', _('Not Selected')),
+        ('MALE', 'M', _('Male')),
+        ('FEMALE', 'F', _('Female')),
+        ('OTHERS', 'O', _('Others')),
+    )
+
+    TSHIRT_SIZE_CHOICES = Choices(
+        ('NOT_SELECTED', 'NA', _('Not Selected')),
+        ('SMALL', 'S', _('Small')),
+        ('MEDIUM', 'M', _('Medium')),
+        ('LARGE', 'L', _('Large')),
+        ('EXTRA_LARGE', 'XL', _('Extra Large')),
+        ('DOUBLE_EXTRA_LARGE', 'XXL', _('Double Extra Large')),
+    )
+
+    first_name = models.CharField(_('First Name'), null=False, blank=True, max_length=120)
+    last_name = models.CharField(_('Last Name'), null=False, blank=True, max_length=120)
     # https://docs.djangoproject.com/en/1.11/ref/contrib/postgres/fields/#citext-fields
-    email = CIEmailField(_('email address'), unique=True, db_index=True)
-    is_staff = models.BooleanField(_('staff status'), default=False,
+    email = CIEmailField(_('Email Address'), null=False, blank=False, unique=True, db_index=True)
+    is_staff = models.BooleanField(_('Staff Status'), default=False, null=False, blank=True,
                                    help_text='Designates whether the user can log into this admin site.')
 
-    is_active = models.BooleanField('active', default=True,
+    is_active = models.BooleanField('Active', default=True, null=False, blank=True,
                                     help_text='Designates whether this user should be treated as '
                                               'active. Unselect this instead of deleting accounts.')
-    date_joined = models.DateTimeField(_('date joined'), default=timezone.now)
+    date_joined = models.DateTimeField(_('Date Of Joining'), default=timezone.now, null=False, blank=False)
 
     USERNAME_FIELD = 'email'
     objects = UserManager()
 
+    gender = models.CharField(_('Gender'), default=GENDER_CHOICES.NOT_SELECTED, choices=GENDER_CHOICES, null=False,
+                              blank=False, max_length=6)
+    tshirt_size = models.CharField(_('Tshirt Size'), default=TSHIRT_SIZE_CHOICES.NOT_SELECTED,
+                                   choices=TSHIRT_SIZE_CHOICES, null=False, blank=False, max_length=10)
+    phone_number = PhoneNumberField(_('Phone Number'), default='', null=False, blank=True, max_length=13)
```

Comment:
Now, including a new plugin just for one Field is quite an issue. We should always try to solve simple problems using the language or framework feature we are provided with. For example, `phone_number` is just a string which only needs to be validated for the correct format of input or phone numbers. I've done it before with just the validator given by Django. Here's the code you might wanna check out and reconsider https://github.com/realslimshanky/Geo-JSON-with-Django/blob/24eed558789a60eb5bf79200c9eb49921f4b4dca/geo_json_with_django/users/models.py#L39

Reply:
It does provide a lot of other utility functions which are good enough. So, let's just use this package.